### PR TITLE
erc20 - cleanup

### PR DESCRIPTION
**Pull request by maaktweluit**
- more logs
 - 3 required confirmations
 - fixed required confirmations
 - fixed faucet early exit
 - tweaked cli timeouts

### CODE REVIEW

File: core/payment-driver/erc20/src/erc20/ethereum.rs [hunk 1]
```diff
@@ -164,7 +164,14 @@ pub async fn is_tx_confirmed(
...
         if let Some(b) = tx.block_number {
-            if b + env.required_confirmations >= *current_block {
+            // TODO: Store tx.block_number in DB and check only once after required_confirmations.
+            log::trace!(
+                "is_tx_confirmed? tb + rq <= cb. tb={}, rq={}, cb={}",
+                b,
+                env.required_confirmations,
+                current_block
+            );
+            if b + env.required_confirmations <= *current_block {
```
Comment (Wiezzel) [inline core/payment-driver/erc20/src/erc20/ethereum.rs:174]: I'm not sure it there shouldn't be `+1` or `-1` somewhere here. When I'm testing it seems like the driver expects one confirmation too many.

I ran `payment_api` with `ERC20_RINKEBY_REQUIRED_CONFIRMATIONS=1`. The first block logged after submitting transaction is `8296279` but the transaction is not confirmed until `8296281` appears. So two blocks, not one.
```
[2021-03-25T13:06:39Z INFO  ya_erc20_driver::driver::cron] Processing payments. count=1, network=rinkeby node_id=0x5ece07692f580f32ea96c7381d7b90fde48e4d36
[2021-03-25T13:06:39Z DEBUG web3::transports::http] [1] Sending: {"jsonrpc":"2.0","method":"eth_getTransactionCount","params":["0x5ece07692f580f32ea96c7381d7b90fde48e4d36","latest"],"id":1} to http://geth.testnet.golem.network:55555/
[2021-03-25T13:06:39Z DEBUG ya_erc20_driver::driver::cron] Payments: nonce=2, details=[PaymentEntity { order_id: "4940ab8f-76c0-4001-b376-45b159bd279c", amount: "0000000000000000000000000000000000000000000000001111f06bcec46380", gas: "0000000000000000000000000000000000000000000000000000000000000000", sender: "0x5ece07692f580f32ea96c7381d7b90fde48e4d36", recipient: "0x946a71f49edf1c5267f7fe2f8563b2940541ea12", payment_due_date: 2021-03-25T13:06:33.412601589, status: 1, tx_id: None, network: Rinkeby }]
[2021-03-25T13:06:39Z DEBUG ya_erc20_driver::erc20::wallet] make_transfer(). network=rinkeby, nonce=2, details=PaymentDetails { recipient: "0x946a71f49edf1c5267f7fe2f8563b2940541ea12", sender: "0x5ece07692f580f32ea96c7381d7b90fde48e4d36", amount: BigDecimal("1.23002851907"), date: Some(2021-03-25T13:06:39.845439329Z) }
[2021-03-25T13:06:39Z DEBUG web3::transports::http] [1] Sending: {"jsonrpc":"2.0","method":"eth_gasPrice","params":[],"id":1} to http://geth.testnet.golem.network:55555/
[2021-03-25T13:06:39Z DEBUG ya_erc20_driver::driver::cron] transactions: [TransactionEntity { tx_id: "2b41bdcec1e8780040a3a5b95cd698ba3e3a3a8552c1c46c186603a6c764056dc0a02e9256026da1e12e093cedd0f2bdd4844c900c888db2b3180599ffeed9f7", sender: "0x5ece07692f580f32ea96c7381d7b90fde48e4d36", nonce: "0000000000000000000000000000000000000000000000000000000000000002", timestamp: 2021-03-25T13:06:39.974180380, status: 1, tx_type: 1, encoded: "{\"nonce\":\"0x2\",\"to\":\"0xd94e3dc39d4cad1dad634e7eb585a57a19dc7efe\",\"value\":\"0x0\",\"gasPrice\":\"0x3b9aca00\",\"gas\":\"0xd6d8\",\"data\":[169,5,156,187,0,0,0,0,0,0,0,0,0,0,0,0,148,106,113,244,158,223,28,82,103,247,254,47,133,99,178,148,5,65,234,18,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,17,17,240,107,206,196,99,128]}", signature: "0016e0344942f02d4bd42f57f07305ce6b91423fedb55a706a22eaae44d056e1115e93ed4928bc4192f04af77dd2e6d1744f492ceb45d1a59439a2eb576b762bd2", tx_hash: None, network: Rinkeby }]
[2021-03-25T13:06:39Z DEBUG web3::transports::http] [1] Sending: {"jsonrpc":"2.0","method":"eth_sendRawTransaction","params":["0xf8a802843b9aca0082d6d894d94e3dc39d4cad1dad634e7eb585a57a19dc7efe80b844a9059cbb000000000000000000000000946a71f49edf1c5267f7fe2f8563b2940541ea120000000000000000000000000000000000000000000000001111f06bcec463802ba016e0344942f02d4bd42f57f07305ce6b91423fedb55a706a22eaae44d056e111a05e93ed4928bc4192f04af77dd2e6d1744f492ceb45d1a59439a2eb576b762bd2"],"id":1} to http://geth.testnet.golem.network:55555/
[2021-03-25T13:06:40Z INFO  ya_erc20_driver::erc20::wallet] Send transaction. hash=0xb12b151559126e02a39a8d181affcbcf53c777a3758ccc3b62dc14817fd1e2b7
[2021-03-25T13:06:40Z DEBUG ya_erc20_driver::erc20::wallet] id=2b41bdcec1e8780040a3a5b95cd698ba3e3a3a8552c1c46c186603a6c764056dc0a02e9256026da1e12e093cedd0f2bdd4844c900c888db2b3180599ffeed9f7
[2021-03-25T13:06:40Z DEBUG ya_erc20_driver::driver::cron] transactions sent!
[2021-03-25T13:06:44Z DEBUG web3::transports::http] [1] Sending: {"jsonrpc":"2.0","method":"eth_blockNumber","params":[],"id":1} to http://geth.testnet.golem.network:55555/
[2021-03-25T13:06:44Z DEBUG ya_erc20_driver::driver::cron] Checking if tx was a success. network=rinkeby, block=8296279, hash=0xb12b151559126e02a39a8d181affcbcf53c777a3758ccc3b62dc14817fd1e2b7
[2021-03-25T13:06:44Z DEBUG web3::transports::http] [1] Sending: {"jsonrpc":"2.0","method":"eth_getTransactionReceipt","params":["0xb12b151559126e02a39a8d181affcbcf53c777a3758ccc3b62dc14817fd1e2b7"],"id":1} to http://geth.testnet.golem.network:55555/
[2021-03-25T13:06:49Z DEBUG web3::transports::http] [1] Sending: {"jsonrpc":"2.0","method":"eth_blockNumber","params":[],"id":1} to http://geth.testnet.golem.network:55555/
[2021-03-25T13:06:49Z DEBUG ya_erc20_driver::driver::cron] Checking if tx was a success. network=rinkeby, block=8296279, hash=0xb12b151559126e02a39a8d181affcbcf53c777a3758ccc3b62dc14817fd1e2b7
[2021-03-25T13:06:49Z DEBUG web3::transports::http] [1] Sending: {"jsonrpc":"2.0","method":"eth_getTransactionReceipt","params":["0xb12b151559126e02a39a8d181affcbcf53c777a3758ccc3b62dc14817fd1e2b7"],"id":1} to http://geth.testnet.golem.network:55555/
[2021-03-25T13:06:54Z DEBUG web3::transports::http] [1] Sending: {"jsonrpc":"2.0","method":"eth_blockNumber","params":[],"id":1} to http://geth.testnet.golem.network:55555/
[2021-03-25T13:06:54Z DEBUG ya_erc20_driver::driver::cron] Checking if tx was a success. network=rinkeby, block=8296280, hash=0xb12b151559126e02a39a8d181affcbcf53c777a3758ccc3b62dc14817fd1e2b7
[2021-03-25T13:06:54Z DEBUG web3::transports::http] [1] Sending: {"jsonrpc":"2.0","method":"eth_getTransactionReceipt","params":["0xb12b151559126e02a39a8d181affcbcf53c777a3758ccc3b62dc14817fd1e2b7"],"id":1} to http://geth.testnet.golem.network:55555/
[2021-03-25T13:06:59Z DEBUG web3::transports::http] [1] Sending: {"jsonrpc":"2.0","method":"eth_blockNumber","params":[],"id":1} to http://geth.testnet.golem.network:55555/
[2021-03-25T13:06:59Z DEBUG ya_erc20_driver::driver::cron] Checking if tx was a success. network=rinkeby, block=8296280, hash=0xb12b151559126e02a39a8d181affcbcf53c777a3758ccc3b62dc14817fd1e2b7
[2021-03-25T13:06:59Z DEBUG web3::transports::http] [1] Sending: {"jsonrpc":"2.0","method":"eth_getTransactionReceipt","params":["0xb12b151559126e02a39a8d181affcbcf53c777a3758ccc3b62dc14817fd1e2b7"],"id":1} to http://geth.testnet.golem.network:55555/
[2021-03-25T13:07:04Z DEBUG web3::transports::http] [1] Sending: {"jsonrpc":"2.0","method":"eth_blockNumber","params":[],"id":1} to http://geth.testnet.golem.network:55555/
[2021-03-25T13:07:04Z DEBUG ya_erc20_driver::driver::cron] Checking if tx was a success. network=rinkeby, block=8296280, hash=0xb12b151559126e02a39a8d181affcbcf53c777a3758ccc3b62dc14817fd1e2b7
[2021-03-25T13:07:04Z DEBUG web3::transports::http] [1] Sending: {"jsonrpc":"2.0","method":"eth_getTransactionReceipt","params":["0xb12b151559126e02a39a8d181affcbcf53c777a3758ccc3b62dc14817fd1e2b7"],"id":1} to http://geth.testnet.golem.network:55555/
[2021-03-25T13:07:09Z DEBUG web3::transports::http] [1] Sending: {"jsonrpc":"2.0","method":"eth_blockNumber","params":[],"id":1} to http://geth.testnet.golem.network:55555/
[2021-03-25T13:07:09Z DEBUG ya_erc20_driver::driver::cron] Checking if tx was a success. network=rinkeby, block=8296281, hash=0xb12b151559126e02a39a8d181affcbcf53c777a3758ccc3b62dc14817fd1e2b7
[2021-03-25T13:07:09Z DEBUG web3::transports::http] [1] Sending: {"jsonrpc":"2.0","method":"eth_getTransactionReceipt","params":["0xb12b151559126e02a39a8d181affcbcf53c777a3758ccc3b62dc14817fd1e2b7"],"id":1} to http://geth.testnet.golem.network:55555/
[2021-03-25T13:07:09Z DEBUG ya_erc20_driver::erc20::wallet] verify_tx. hash=0xb12b151559126e02a39a8d181affcbcf53c777a3758ccc3b62dc14817fd1e2b7
[2021-03-25T13:07:09Z DEBUG web3::transports::http] [1] Sending: {"jsonrpc":"2.0","method":"eth_getTransactionReceipt","params":["0xb12b151559126e02a39a8d181affcbcf53c777a3758ccc3b62dc14817fd1e2b7"],"id":1} to http://geth.testnet.golem.network:55555/
[2021-03-25T13:07:10Z DEBUG ya_erc20_driver::erc20::wallet] PaymentDetails from server: PaymentDetails { recipient: "0x946a71f49edf1c5267f7fe2f8563b2940541ea12", sender: "0x5ece07692f580f32ea96c7381d7b90fde48e4d36", amount: BigDecimal("1.23002851907"), date: Some(2021-03-25T13:07:10.072162961Z) }
[2021-03-25T13:07:10Z DEBUG ya_erc20_driver::driver::api] verify_payment: VerifyPayment { confirmation: PaymentConfirmation { confirmation: [177, 43, 21, 21, 89, 18, 110, 2, 163, 154, 141, 24, 26, 255, 203, 207, 83, 199, 119, 163, 117, 140, 204, 59, 98, 220, 20, 129, 127, 209, 226, 183] }, platform: "erc20-rinkeby-tglm" }
[2021-03-25T13:07:10Z INFO  ya_erc20_driver::driver::api] Verifying transaction: 0xb12b151559126e02a39a8d181affcbcf53c777a3758ccc3b62dc14817fd1e2b7
[2021-03-25T13:07:10Z DEBUG ya_erc20_driver::erc20::wallet] verify_tx. hash=0xb12b151559126e02a39a8d181affcbcf53c777a3758ccc3b62dc14817fd1e2b7
[2021-03-25T13:07:10Z DEBUG web3::transports::http] [1] Sending: {"jsonrpc":"2.0","method":"eth_getTransactionReceipt","params":["0xb12b151559126e02a39a8d181affcbcf53c777a3758ccc3b62dc14817fd1e2b7"],"id":1} to http://geth.testnet.golem.network:55555/
[2021-03-25T13:07:10Z DEBUG ya_erc20_driver::erc20::wallet] PaymentDetails from server: PaymentDetails { recipient: "0x946a71f49edf1c5267f7fe2f8563b2940541ea12", sender: "0x5ece07692f580f32ea96c7381d7b90fde48e4d36", amount: BigDecimal("1.23002851907"), date: Some(2021-03-25T13:07:10.303720160Z) }
```

**Review by Wiezzel — CHANGES_REQUESTED** [review]
1. Seems there's something wrong with required number of confirmations.
2. Please check the CI integration test failure (might be related to 1).
3. Please update `.env-template`:
```
ERC20_RINKEBY_REQUIRED_CONFIRMATIONS=1
```

File: core/payment-driver/erc20/src/erc20/ethereum.rs (hunk 1 above)
Comment (maaktweluit) [reply core/payment-driver/erc20/src/erc20/ethereum.rs]: good catch! it was one block too late.. fixed :)

**Comment by maaktweluit** [comment]
1. replied above, good catch
2. the master is also failing, i think its related to the `GNT` removal, updating the `zk-mock` image in goth
3. fixed as well

**Review by Wiezzel — APPROVED** [review]

**Review by Wiezzel — DISMISSED** [review]

**Review by mfranciszkiewicz — DISMISSED** [review]

**Comment by maaktweluit** [comment]
Builds did not trigger :(
Had to dismiss the reviews...

Will try and get this still merged today since its broken on master now

**Review by mfranciszkiewicz — DISMISSED** [review]

**Review by Wiezzel — APPROVED** [review]

**Pull request merged**